### PR TITLE
Chakra UI

### PR DESCRIPTION
**Pull request by Wei-Xinn**
## Main Change
This PR revamps the existing frontend code to use Chakra UI instead of Material UI, along with the following

## Other changes
- QuestionString to use string[ ]  instead of string for categories
- Fix table cell overflow styling issue
- Change Notification System (Handler to React Hook)

### CODE REVIEW

**Review by dsja612 — DISMISSED** [review]
Small nitpicks as discussed:

1. categories formatting in QuestionDetailsModal:
image

2. After adding a question, add question page starts at ModalPage2 
image

**Review by dsja612 — APPROVED** [review]
LGTM, good job

File: frontend/src/theme.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { extendTheme, type ThemeConfig } from '@chakra-ui/react'
+
+const config: ThemeConfig = {
+  initialColorMode: 'dark',
+  useSystemColorMode: false,
+}
+
+const theme = extendTheme({ config })
```
Comment (sharpstorm) [inline frontend/src/theme.tsx]:
```suggestion
const theme = extendTheme({
  config,
  styles: {
    global: () => ({
      body: {
        bg: '#282c34',
      }
    })
  }
})
```

File: frontend/src/MockData.tsx [hunk 2]
```diff
@@ -44,10 +46,11 @@ const desc2 =
 
 
 const mockQuestions = [
```
Comment (sharpstorm) [inline frontend/src/MockData.tsx]:
```suggestion
const mockQuestions: QuestionString[] = [
```

File: frontend/src/Util.tsx [hunk 3]
```diff
@@ -13,4 +13,17 @@ function getCategoriesString() {
...
+  if (input.length === 0) {
+    return [];
+  }
+  let arr = input.split(', ').map((value: string) => {
+    return ({
+      value: value,
+      label: value
+    });
+  });
+  return arr;
```
Comment (sharpstorm) [inline frontend/src/Util.tsx]:
```suggestion
return input.split(', ').map((value: string) => ({
    value: value,
    label: value
  }));
```

File: frontend/src/pages/QuestionPage.css [hunk 4]
```diff
@@ -11,7 +11,6 @@
   justify-content: flex-end;
 }
 
-#add-btn,
-#delete-btn {
+#add-btn {
```
Comment (sharpstorm) [inline frontend/src/pages/QuestionPage.css:14]: Would be better to delete the CSS file, use emotion

File: frontend/src/components/question/ManageQuestionsButtonRow.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { Button } from '@chakra-ui/react'
+
+interface Props {
+  addHandler: () => void,
+}
+
+const ManageQuestionsButtonRow: React.FC<Props> = ({ addHandler }) => {
+  return (
+    <div id='button-container'>
+      <Button id='add-btn' colorScheme='blue' onClick={addHandler}>
```
Comment (sharpstorm) [inline frontend/src/components/question/ManageQuestionsButtonRow.tsx]:
```suggestion
      <Button m="5px" colorScheme='blue' onClick={addHandler}>
```

File: frontend/src/components/question/QuestionTable.tsx [hunk 6]
```diff
@@ -2,28 +2,17 @@ import './QuestionTable.css';
 import { useState, useEffect } from "react";
 import Question from '../../models/Question';
 import { QuestionString } from '../../Commons';
-import ManageQuestionsButtonRow from '../ManageQuestionsButtonRow';
+import ManageQuestionsButtonRow from './ManageQuestionsButtonRow';
+import { TableContainer, Table, Thead, Tr, Th, Tbody, Td } from '@chakra-ui/react';
```
Comment (sharpstorm) [inline frontend/src/components/question/QuestionTable.tsx:5]: consider:
```suggestion
import { TableContainer, Table, Thead, Tr, Th, Tbody, Td } from '@chakra-ui/react';

const headings = ['Id', 'Title', 'Category', 'Complexity'];
```

File: frontend/src/components/question/QuestionTable.tsx [hunk 7]
```diff
@@ -35,45 +24,36 @@ const QuestionTable: React.FC<Props> =
...
-              {headerCell("Complexity")}
-              {headerCell("Link")}
-            </tr>
+        <Table variant='simple' className='question-table' width={'70vw'}>
+          <Thead>
+            <Tr>
+              <Th className='question-th'>Id</Th>
+              <Th className='question-th'>Title</Th>
+              <Th className='question-th'>Category</Th>
+              <Th className='question-th'>Complexity</Th>
```
Comment (sharpstorm) [inline frontend/src/components/question/QuestionTable.tsx]: Consider:
```suggestion
              headings.map((label) => <Th key={label} className='question-th'>{label}</Th>)
```

File: frontend/src/components/question/addModal/AddQuestionModal.tsx [hunk 8]
```diff
@@ -1,86 +1,82 @@
...
   submitHandler: () => void;
 }
 
+const ModalButton = (label: string, onClick: () => void) => {
+  return (
+    <Button colorScheme='blue' mr={3} onClick={onClick}>
+      {label}
+    </Button>
+  );
+};
```
Comment (sharpstorm) [inline frontend/src/components/question/addModal/AddQuestionModal.tsx]:
```suggestion
const ModalButton = ({label, onClick} : {label: string, onClick: () => void}) => {
  return (
    <Button colorScheme='blue' mr={3} onClick={onClick}>
      {label}
    </Button>
  );
};
```

File: frontend/src/components/question/addModal/AddQuestionModal.tsx [hunk 9]
```diff
@@ -1,86 +1,82 @@
...
 const AddQuestionModal: React.FC<Props> =
-  ({ isVisible, closeHandler, newQuestionSetter, submitHandler }) => {
+  ({ isVisible, closeHandler, submitHandler }) => {
+    const [page, setPage] = useState(1);
+    const { setQuestionData } = useContext(NewQuestionContext);
+    const SecondPageButtons = () => {
+      return (
+        <>
+          {ModalButton('Previous', () => { setPage(1) })}
+          {ModalButton('Submit', () => { try { submitHandler() } catch { close(); } })}
```
Comment (sharpstorm) [inline frontend/src/components/question/addModal/AddQuestionModal.tsx]:
```suggestion
          <ModalButton label="Previous" onClick={() => { setPage(1) }} />
          <ModalButton
            label="Submit"
            onClick={() => {
              try {
                submitHandler();
              } catch {
                close();
              }
            } />
```

File: frontend/src/components/question/addModal/AddQuestionModal.tsx [hunk 10]
```diff
@@ -1,86 +1,82 @@
...
+          <ModalBody>
+            {page === 1 ?
+              <ModalPage1 /> :
+              <ModalPage2 />
+            }
+          </ModalBody>
+          <ModalFooter>
+            {
+              page === 1 ?
+                ModalButton('Next', () => { setPage(2) })
```
Comment (sharpstorm) [inline frontend/src/components/question/addModal/AddQuestionModal.tsx]:
```suggestion
                <ModalButton label="Next" onClick={() => setPage(2)} />
```

File: frontend/src/contexts/NewQuestionContext.tsx [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+import { QuestionString, questionStringTemplate } from "../Commons";
+
+interface ContextProps {
+  questionData: QuestionString,
+  setQuestionData: (newData: QuestionString) => void
+}
+
+export const NewQuestionContext = createContext<ContextProps>({
+  questionData: questionStringTemplate,
+  setQuestionData: () => null
```
Comment (sharpstorm) [inline frontend/src/contexts/NewQuestionContext.tsx]:
```suggestion
  setQuestionData: () => {}
```

File: frontend/src/App.css [hunk 12]
```diff
@@ -1,6 +1,3 @@
-#root {
-  max-width: 1280px;
-  margin: 0 auto;
-  padding: 2rem;
-  text-align: center;
+body {
+  background-color: #282c34 !important;
```
Comment (Wei-Xinn) [inline frontend/src/App.css]: fixed

**Review by LeirdaKJW — COMMENTED** [review]
Adding a question still causes the modal page to start from 2:
image

**Pull request merged**